synapse/events/utils.py

```diff
@@ -368,11 +369,24 @@ def serialize_event(
             e.unsigned["redacted_because"], time_now_ms, config=config
         )
 
-    if config.token_id is not None:
-        if config.token_id == getattr(e.internal_metadata, "token_id", None):
-            txn_id = getattr(e.internal_metadata, "txn_id", None)
-            if txn_id is not None:
-                d["unsigned"]["transaction_id"] = txn_id
+    # If we have a txn_id saved in the internal_metadata, we should include it in the
+    # unsigned section of the event if it was sent by the same session as the one
+    # requesting the event.
+    # There is a special case for guests, because they only have one access token
+    # without associated access_token_id, so we always include the txn_id for events
+    # they sent.
+    txn_id = getattr(e.internal_metadata, "txn_id", None)
```

Comment:
I think I can do:

```suggestion
    txn_id = e.internal_metadata.txn_id
```

At least it looks like it from how it is defined:

https://github.com/matrix-org/synapse/blob/db2b105d69fa331bb3f050df82266314f61577ea/synapse/events/__init__.py#L201